week0003/daudau/php/daudau.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+$bank = Bank::getInstance();
+$bank->welcome();
+
+class Account
+{
+	private $_name;
+	private $_money;
+
+	function __construct($name, $money) {
+		$this->_name = $name;
+		$this->_money = $money;
+	}
+
+	public function withdraw($amount) {
+		if ($amount < 0 || $amount > $this->_money)
+			return false;
+		$this->_money -= $amount;
+		return true;
+	}
+
+	public function deposite($amount) {
+		if ($amount < 0)
+			return false;
+		$this->_money += $amount;
+		return true;
+	}
+
+	public function save() {
+		$bank = Bank::getInstance();
+		$bank->accounts()[strtolower($this->getName())] = $this;
+	}
+
+	public function getName() {
+		return $this->_name;
+	}
+
+	public function getMoney() {
+		return $this->_money;
+	}
+}
+
+/**
+* using bank instance
+*/
+class Bank
+{
+	private $_accounts;
+	private $_keywords;
+    private static $_instance;
+
+    protected function __construct() {
+		$this->_accounts = array();
+		$this->_keywords = array('A', 'D', 'W');
+	}
+
+	public static function getInstance() {
+	    if (null === self::$_instance) {
+            self::$_instance = new self();
+        }
+        return self::$_instance;
+    }
+
+	public function add(Account $account) {
+		if ( $this->_accounts[strtolower($account->getName())] )
+			return false;
+		else
+			$this->_accounts[strtolower( $account->getName() )] = $account;
+		return true;
+	}
+
+	public function save($account) {
+		if ( $this->_accounts[strtolower($account->getName())] ) {
+			$this->_accounts[strtolower($account->getName())] = $account;
+			return true;
+		}
+		return false;
+	}
+
+	public function welcome() {
+		while (1) {
```

Comment:
You read a line, process it, print it, process next line, repeat, until you get the end-of-file